File: apps/runtime/nodejs/generate-manifest.mjs
// Generates runtime-manifest.json listing installed @celerity-sdk/* versions.
// Run during Docker build: node generate-manifest.mjs

import { readFileSync, readdirSync, writeFileSync } from "node:fs";
import { join } from "node:path";

function generateManifest() {
  const sdkDir = "./node_modules/@celerity-sdk";
  const manifest = { generatedAt: new Date().toISOString(), sdkVersions: {} };

  try {
    for (const pkg of readdirSync(sdkDir)) {
      try {
        const pkgJson = JSON.parse(
          readFileSync(join(sdkDir, pkg, "package.json"), "utf8"),
        );
        manifest.sdkVersions[pkg] = pkgJson.version;
      } catch {
        // Skip platform-specific binaries without a standard package.json
      }
    }
  } catch {
    // @celerity-sdk directory not found
  }

  writeFileSync("runtime-manifest.json", JSON.stringify(manifest, null, 2));
}

generateManifest();

File: apps/runtime/python/generate_manifest.py
# Generates runtime-manifest.json listing installed celerity SDK versions.
# Run during Docker build: python generate_manifest.py

from __future__ import annotations

import contextlib
import json
from datetime import UTC, datetime
from importlib.metadata import PackageNotFoundError, version

SDK_PACKAGES = ["celerity-sdk", "celerity-runtime-sdk"]


def generate_manifest() -> None:
    sdk_versions: dict[str, str] = {}
    for pkg in SDK_PACKAGES:
        with contextlib.suppress(PackageNotFoundError):
            sdk_versions[pkg] = version(pkg)

    manifest = {
        "generatedAt": datetime.now(UTC).isoformat(),
        "sdkVersions": sdk_versions,
    }

    with open("runtime-manifest.json", "w") as f:
        json.dump(manifest, f, indent=2)


if __name__ == "__main__":
    generate_manifest()
